test/runtests.jl

```diff
@@ -163,6 +163,55 @@ mkpath(TEST_EXAMPLES_DIR)
       @test sum(data) ≈ 192.1204941112099
     end
   end
+
+  # Test for validation of uniform grid ("image data") read feature
+  @testset "ImageData" begin
+
+    ## Generate grid file and write vti
+    
+    # grid geometry parameter
+    input_origin  = [ 1.0, 1.0, 2.0 ]
+    input_spacing = [ 1.0, 1.0, 0.1 ]
+    input_ending  = [ 3.0, 2.0, 2.2 ]
+
+    # compute ranges
+    x, y, z = [ (input_origin[i] : input_spacing[i] : input_ending[i])  for i in (1:3) ]
+    Nx, Ny, Nz = length(x), length(y), length(z)
+
+    # generate random data
+    point_scalar_field = rand(Nx, Ny, Nz)
+    point_data_name    = "Point scalar data"
+
+    cell_scalar_field  = rand(Nx-1, Ny-1, Nz-1)
+    cell_data_name     = "Name of Test Cell scalar data"
+
+    # write vti file using WriteVTK
+    vtk_grid("grid", x, y, z) do vtk
+        vtk[ point_data_name, VTKPointData()] = point_scalar_field  # scalar field attached to points
+        vtk[ cell_data_name,  VTKCellData() ] = cell_scalar_field   # scalar field attached to cells
+    end
+    
+    ## Read vti file using ReadVTK
+    filepath = "grid.vti"
+    vtk = VTKFile( filepath )
+    
+    origin  = get_origin( vtk )
+    spacing = get_spacing( vtk )
+    cell_data_read = get_data( get_cell_data(vtk)[cell_data_name] )
+    point_data_read = get_data( get_point_data(vtk)[point_data_name] )
+    
+    cell_data_read = reshape( cell_data_read,    ( (Nx-1), (Ny-1), (Nz-1) ) )
+    point_data_read = reshape( point_data_read,  (     Nx,     Ny,     Nz ) )
+
+    # test if cell and point data are well read
+    @test cell_data_read == cell_scalar_field
+    @test point_data_read == point_scalar_field
+
+    # test if the origin and spacing are well read
+    @test origin  == input_origin
+    @test spacing == input_spacing
```

Comment:
I would slightly re-arrange the tests to make it easier to spot where the problem is in case of an error:
```suggestion
    # Read vti file using ReadVTK
    filepath = "grid.vti"
    @testset "VTKFile" begin
      @test VTKFile(filepath) isa VTKFile
    end
    vtk = VTKFile(filepath)

    # check getter functions
    @testset "get_origin" begin
      @test get_origin(vtk) == input_origin
    end

    @testset "get_spacing" begin
      @test get_spacing(vtk) == input_spacing
    end

    @testset "get cell data" begin
      cell_data_raw = get_data(get_cell_data(vtk)[cell_data_name])
      cell_data_reshaped = reshape(cell_data_raw, ((Nx-1), (Ny-1), (Nz-1)))
      @test cell_data_reshaped == cell_scalar_field
    end

    @testset "get point data" begin
      point_data_raw = get_data(get_point_data(vtk)[point_data_name])
      point_data_reshaped = reshape(point_data_raw, (Nx, Ny, Nz))
      @test point_data_reshaped == point_scalar_field
    end
```